src/jit/codegenarm64.cpp

```diff
@@ -5059,9 +5059,104 @@ void CodeGen::genHWIntrinsicSimdBinaryOp(GenTreeHWIntrinsic* node)
     genProduceReg(node);
 }
 
+template <typename HWIntrinsicSwitchCaseBody>
```

Comment:
This needs a function header. See https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#94-function-header-comment

I would in particular describe the scenario this supports, as it may be confusing at first why we have a HW intrinsic switch.